pennylane/transforms/add_noise.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Transform for adding a noise model to a quantum circuit or device"""
+from copy import copy
+from functools import lru_cache
+
+import pennylane as qml
+from pennylane.transforms.core import TransformContainer, transform
+
+
+@transform
+def add_noise(tape, noise_model, level=None):
+    """Insert operations according to a provided noise model.
+
+    Circuits passed through this transform will be updated to have the operation,
+    specified by the ``noise_model`` argument after appyling the transforms specified
```

Comment:
as a user I would not know that this exists in this context. Could you include some reference inside the `qml.noise` module documentation?

Reply:
Added!